FormalConjectures/ErdosProblems/503.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 503
+
+*Reference:* [erdosproblems.com/503](https://www.erdosproblems.com/503)
+-/
+
+namespace Erdos503
+
+abbrev ESpace (n : ℕ) := EuclideanSpace ℝ (Fin n)
+
+def isIsosceles {n : ℕ} (x y z : ESpace n) :=
+  dist x y = dist y z ∨ dist y z = dist x z ∨ dist x y = dist x z
```

Comment:
How about moving these definitions to a new file FormalConjectures/ForMathlib/Geometry/Euclidean.lean or something similar? There are more problems that use "isosceles" (https://www.erdosproblems.com/search/isosceles), and having a notation for $$\mathbb{R}^n$$ handy is also a good idea